NumLib/ODESolver/ODETypes.h

```diff
@@ -0,0 +1,210 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_ODETYPES_H
+#define NUMLIB_ODETYPES_H
+
+// TODO: move that file somewhere else
+
+#include <initializer_list>
+#include <cassert>
+
+
+// Dense Eigen matrix/vector //////////////////////////////////////////
+// always enabled
+
+#include <Eigen/LU>
+
+
+inline void setMatrix(Eigen::MatrixXd& m,
+                      Eigen::MatrixXd::Index const rows, Eigen::MatrixXd::Index const cols,
+                      std::initializer_list<double> values)
+{
+    using IndexType = Eigen::MatrixXd::Index;
+    assert((IndexType) values.size() == rows*cols);
+
+    auto it = values.begin();
+    for (IndexType r=0; r<rows; ++r) {
+        for (IndexType c=0; c<cols; ++c) {
+            m(r, c) = *(it++);
+        }
+    }
+}
+
+inline void setMatrix(Eigen::MatrixXd& m, Eigen::MatrixXd const& tmp)
+{
+    m = tmp;
+}
+
+inline void addToMatrix(Eigen::MatrixXd& m,
+                        Eigen::MatrixXd::Index const rows, Eigen::MatrixXd::Index const cols,
+                        std::initializer_list<double> values)
+{
+    using IndexType = Eigen::MatrixXd::Index;
+    assert((IndexType) values.size() == rows*cols);
+
+    auto it = values.begin();
+    for (IndexType r=0; r<rows; ++r) {
+        for (IndexType c=0; c<cols; ++c) {
+            m(r, c) += *(it++);
+        }
+    }
+}
+
+namespace NumLib
```

Comment:
Actually it would be better to put all of this into the MathLib namespace in the end, because all matrices, vectors and linear solvers reside there.
However, I will put everything into the NumLib namespace since the file is located there. I will add a comment that ths will have to change.